pkg/openstack/rabbitmq.go

```diff
@@ -37,7 +38,28 @@ func ReconcileRabbitMQ(ctx context.Context, instance *corev1beta1.OpenStackContr
 			Template: &rabbitmqv1.PodTemplateSpec{
 				Spec: &corev1.PodSpec{
 					SecurityContext: &corev1.PodSecurityContext{},
-					Containers:      []corev1.Container{},
+					Containers: []corev1.Container{
+						corev1.Container{
+							Name: "rabbitmq",
+							Env: []corev1.EnvVar{
+								corev1.EnvVar{
+									// The upstream rabbitmq image has /var/log/rabbitmq mode 777, so when
+									// openshift runs the rabbitmq container as a random uid it can still write
+									// the logs there.  The OSP image however has the directory more constrained,
+									// so the random uid cannot write the logs there.  Force it into /var/lib
+									// where it can create the file without crashing.
+									Name:  "RABBITMQ_UPGRADE_LOG",
+									Value: "/var/lib/rabbitmq/rabbitmq_upgrade.log",
+								},
+							},
+							Command: []string{
+								// The OSP17 image wants to start with kolla, override it to run rabbitmq directly.
+								// For some reason HOME needs to be explictly set here even though the entry
```

Comment:
Probably due to the entrypoint being set to dumb-init (https://github.com/openstack/tripleo-common/blob/master/container-images/tcib/base/base.yaml#L56)

Reply:
Setting `Command` on the `Pod` overrides the `ENTRYPOINT` supplied in the image (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#container-v1-core).  So this version of the patch actually doesn't use dumb-init at all.  I also tried explicitly passing `-l` to `/bin/sh` to force a login shell, but that still does not populate `$HOME` correctly.

Ultimately I've reorganized this a bit (about to push a new version), split the `HOME` part out into the `EnvVar` array, and keeping the default entrypoint (dumb-init) but overriding the args to run `rabbitmq-server` instead of `kolla_start`.  I think that's cleaner and more idiomatic.